Source/WebCore/svg/SVGURIReference.cpp

```diff
@@ -62,18 +62,19 @@ void SVGURIReference::parseAttribute(const QualifiedName& name, const AtomString
 
 AtomString SVGURIReference::fragmentIdentifierFromIRIString(const String& url, const Document& document)
 {
-    size_t start = url.find('#');
+    auto trimmedURL = url.trim(isASCIIWhitespace<UChar>);
+    size_t start = trimmedURL.find('#');
     if (start == notFound)
         return emptyAtom();
 
     if (!start)
-        return StringView(url).substring(1).toAtomString();
+        return StringView(trimmedURL).substring(1).toAtomString();
 
-    URL base = URL(document.baseURL(), url.left(start));
-    String fragmentIdentifier = url.substring(start);
+    URL base = URL(document.baseURL(), trimmedURL.left(start));
+    String fragmentIdentifier = trimmedURL.substring(start);
```

Comment:
auto?

Reply:
Does not compile but I have plan to do refactoring here later on to simplify this function, I will try to tackle it in that.